src/components/ContextBar.test.js

```diff
@@ -0,0 +1,625 @@
+  diskType: {
+    label: 'pd-standard',
+    displayName: 'Standard',
+    regionToPricesName: 'monthlyStandardDiskPrice'
+  },
+  blockSize: 4096,
+  labels: {
+    saturnWorkspaceNamespace: 'general-dev-billing-account',
+    saturnWorkspaceName: 'Broad Test Workspace'
+  }
+}
+
+const rstudioRuntime = {
+  id: 76979,
+  workspaceId: null,
+  runtimeName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+  googleProject: 'terra-dev-98897219',
+  cloudContext: {
+    cloudProvider: 'GCP',
+    cloudResource: 'terra-dev-98897219'
+  },
+  auditInfo: {
+    creator: 'ncl.hedwig@gmail.com',
+    createdDate: '2022-09-08T19:46:37.396597Z',
+    destroyedDate: null,
+    dateAccessed: '2022-09-08T19:47:21.206Z'
+  },
+  runtimeConfig: {
+    machineType: 'n1-standard-4',
+    persistentDiskId: 15778,
+    cloudService: 'GCE',
+    bootDiskSize: 120,
+    zone: 'us-central1-a',
+    gpuConfig: null
+  },
+  proxyUrl: 'https://leonardo.dsde-dev.broadinstitute.org/proxy/terra-dev-98897219/saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb/rstudio',
+  status: 'Running',
+  labels: {
+    saturnWorkspaceNamespace: 'general-dev-billing-account',
+    'saturn-iframe-extension': 'https://bvdp-saturn-dev.appspot.com/jupyter-iframe-extension.js',
+    creator: 'ncl.hedwig@gmail.com',
+    clusterServiceAccount: 'pet-26534176105071279add1@terra-dev-98897219.iam.gserviceaccount.com',
+    saturnAutoCreated: 'true',
+    clusterName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    saturnWorkspaceName: 'N8s Space',
+    saturnVersion: '6',
+    tool: 'RStudio',
+    runtimeName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    cloudContext: 'Gcp/terra-dev-98897219',
+    googleProject: 'terra-dev-98897219'
+  },
+  patchInProgress: false
+}
+
+const contextBarProps = {
+  runtimes: [],
+  apps: [],
+  appDataDisks: [],
+  refreshRuntimes: () => '',
+  location: 'US-CENTRAL1',
+  locationType: '',
+  refreshApps: () => '',
+  workspace: {
+    workspace: {
+      namespace: 'namespace'
+    },
+    namespace: 'Broad Test Workspace'
+  }
+}
+
+describe('ContextBar - buttons', () => {
+  it('will render default icons', () => {
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, contextBarProps))
+
+    // Assert
+    expect(getByText('Rate:'))
+    expect(getByLabelText('Environment Configuration'))
+    expect(getByLabelText('Terminal button')).toHaveAttribute('disabled')
+  })
+
+  it('will render Jupyter button with an enabled Terminal Button', () => {
+    // Arrange
+    const jupyterContextBarProps = {
+      ...contextBarProps,
+      runtimes: [jupyter],
+      persistentDisks: [runtimeDisk]
+    }
+
+    // Act
+    const { getByText, getByLabelText } = render(h(ContextBar, jupyterContextBarProps))
+
+    //Assert
+    expect(getByText('Rate:'))
+    expect(getByLabelText('Environment Configuration'))
+    expect(getByLabelText(new RegExp(/Jupyter Environment/i)))
+    expect(getByLabelText('Terminal button')).toBeEnabled()
```

Comment:
It looks like this is actually testing more than it claims to ("will render Jupyter button with an enabled Terminal Button"). Might be worth splitting some of this out into separate tests or relabeling the test.